Assets/Scripts/AbstractClasses/Damageable.cs

```diff
@@ -14,15 +14,23 @@ public abstract class Damageable : MonoBehaviour
         private static readonly int DieTrigger = Animator.StringToHash(Consts.Animation.DIE_TRIGGER);
 
         /// <summary>
-        /// Marks the player as hit after being hit.
+        /// Marks the entity as hit after being hit.
         /// </summary>
-        /// <param name="damage">The damage dealt to player</param>
+        /// <param name="damage">The damage dealt</param>
         /// <param name="attacker">The EnemyController of the attacker if it's an NPC.</param>
         public virtual void Hit(int damage, EnemyController attacker = null)
         {
             animator.SetTrigger(HitTrigger);
         }
 
+        /// <summary>
+        /// Does damage to the player. Implementation should call base.Hit(damage, attacker)
+        /// </summary>
+        /// <param name="damage">The damage dealth to the player</param>
+        /// <param name="health">The health of the entity after damage is dealt</param>
+        /// <param name="attacker">The EnemyController of the attacker if it's an NPC.</param>
+        public abstract void Hit(int damage, out int health, EnemyController attacker = null);
```

Comment:
I don't really like the fact that we have two Hit method definitions 😬 Wouldn't it be possible to just alter the existing implementation for your needs?

Reply:
I think this is the only possibility, if you don't want to create new references and coupling.

Reply:
AttackLogic and Hit is not clean anyway, since AttackLogic is used for both enemy and player. When we merge the saturation, hydration stuff that I'm working on, it won't get better unfortunately :(  But I don't have time to refactor it and rewrite the whole AttackLogic.

Reply:
Okay, it's fine then, I guess 🤷‍♂